src/functionalTest/java/uk/gov/hmcts/reform/divorce/ccdcase/PatchCaseInCCDTest.java

```diff
@@ -1,75 +1,99 @@
 
     @Value("${case.orchestration.maintenance.case.context-path}")
-    private String draftCaseContextPath;
+    private String casePath;
 
     @Test
-    public void givenDivorceSessionWithId_whenPatchIsCalled_caseIsUpdated() throws Exception {
-        UserDetails citizenUser = createCitizenUser();
+    public void shouldPatchCaseFields() throws Exception {
 
-        final Response response = submitDraftCase(citizenUser, DRAFT_DIVORCE_SESSION_JSON_PATH);
-        final String caseId = response.getBody().path(CASE_ID_JSON_KEY);
+        final UserDetails citizenUser = createCitizenUser();
+
+        final Response response = submitDraftCase(citizenUser, "initial-case-for-patch.json");
         assertThat(response.getStatusCode(), is(OK.value()));
 
-        final String patchWithIdJson =
-            ResourceLoader.loadJson(PAYLOAD_CONTEXT_PATH + PATCH_DIVORCE_SESSION_JSON_PATH)
-                .replace("caseId", caseId);
+        final String caseId = response.getBody().path(CASE_ID_JSON_KEY);
+        final String patchWithIdJson = loadJson(PAYLOAD_CONTEXT_PATH + "patch-case.json")
+            .replace("caseId", caseId);
+
+        final Response beforePatchResponse = getCase(citizenUser.getAuthToken());
+        assertThat(beforePatchResponse.getStatusCode(), is(OK.value()));
+        assertThat(beforePatchResponse.getBody().path("state"), is("Draft"));
+        assertThat(beforePatchResponse.getBody().path("data.D8ScreenHasMarriageBroken"), is("YES"));
+        assertThat(beforePatchResponse.getBody().path("data.D8PetitionerFirstName"), nullValue());
+        assertThat(beforePatchResponse.getBody().path("data.D8PetitionerLastName"), nullValue());
+        assertThat(beforePatchResponse.getBody().path("data.D8DerivedPetitionerCurrentFullName"), nullValue());
+        assertThat(beforePatchResponse.getBody().path("data.D8FinancialOrderFor"), contains("petitioner", "children"));
 
         final Response patchResponse = patchCase(citizenUser, patchWithIdJson);
-
-        assertEquals(HttpStatus.OK.value(), patchResponse.getStatusCode());
+        assertThat(patchResponse.getStatusCode(), is(OK.value()));
+
+        final Response afterPatchResponse = getCase(citizenUser.getAuthToken());
+        assertThat(afterPatchResponse.getStatusCode(), is(OK.value()));
+        assertThat(afterPatchResponse.getBody().path("id"), is(caseId));
+        assertThat(afterPatchResponse.getBody().path("state"), is("Draft"));
+        assertThat(afterPatchResponse.getBody().path("data.D8ScreenHasMarriageBroken"), is("YES"));
+        assertThat(afterPatchResponse.getBody().path("data.D8PetitionerFirstName"), is("John"));
+        assertThat(afterPatchResponse.getBody().path("data.D8PetitionerLastName"), is("Jones"));
+        assertThat(afterPatchResponse.getBody().path("data.D8DerivedPetitionerCurrentFullName"), is("John Jones"));
+        assertThat(afterPatchResponse.getBody().path("data.D8FinancialOrderFor"), contains("petitioner"));
```

Comment:
is there a way to assert the whole json ?

Reply:
I wanted to keep this targeted to only the fields that are being patched, but I have made it more readable by using JsonPath for the assertions.